modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/transformation/EventUtils.scala

```diff
@@ -93,6 +94,27 @@ object EventUtils {
     tabular.mkString("\t")
   }
 
+  def alterEnrichedEventAny[F[_]: Clock: RegistryLookup: Monad](originalLine: Event, lengths: Map[String, Int]): List[Any] = {
```

Comment:
What do you think about using `Field` and `FieldValue` from schema-ddl also for atomic fields? 

For non-atomic entities (unstruct + contexts) we build `Field` from schema and then cast to `FieldValue`. We could also use casting to `FieldValue` for atomic, for that we would need atomic `Field`s definition. These could be defined in code like in `SparkSchema`, sth like: 

```scala 
List(
    Field("app_id",                   Type.String,     Nullable),
    Field("platform",                 Type.String,     Nullable),
...
```
and then use `FieldValue.cast` cause we would have a field and json value. Such atomic fields definition could be in common cause it uses schema-ddl and then be shared by batch and streaming transformers. Does it make sense?

Reply:
I like this idea, if you can get it working.

At one point I considered taking this a step further, and using [the atomic schema definition](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/atomic/jsonschema/1-0-0) to auto-derive the field types.  But I couldn't find a nice way to get that working.

Your suggestion to define the `Field`s explicitly is a nice half-way solution, and much neater and probably more type-safe than what we have at the moment.